### PR TITLE
[WFCORE-5069] Automatically add the -Djboss.modules.os-name property if it has not already been specified when running on RHEL 6, RHEL 7, or RHEL 8 to ensure the correct WildFly OpenSSL native library will be used

### PR DESCRIPTION
**Pull request by fjuma**
https://issues.redhat.com/browse/WFCORE-5069

### CODE REVIEW

File: launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java [hunk 1]
```diff
@@ -587,6 +606,35 @@ protected Path normalizePath(final Path parent, final String path) {
         return serverArgs.asList();
     }
 
+    /**
+     * Returns the value of the {@code jboss.modules.os-name} property.
+     *
+     * @return the value of the {@code jboss.modules.os-name} property
+     */
+    public String getJBossModulesOSName() {
```
Comment (jmesnil) [inline launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java:614]: I don't understand this patch.
It seems we are duplicating the logic to determine the `jbossModulesOSName` in both the scripts and here. Am I missing a difference?

Wouldn't it be enough to have only this `getJBossModulesOSName`method that will automatically set the field if the `os.name` correspond to RHEL 6-8?

Comment (jstourac) [reply launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java:614]: Also again, I am a bit worried that there will be a problem with WildFly distro executed on the RHELX platform - If I didn't miss anything then with this modification we fix this for EAP, although user of WildFly will now have to set `jboss.modules.os-name` with proper value in such case, right?

Comment (fjuma) [reply launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java:614]: > I don't understand this patch.
> It seems we are duplicating the logic to determine the `jbossModulesOSName` in both the scripts and here. Am I missing a difference?
> 
> Wouldn't it be enough to have only this `getJBossModulesOSName`method that will automatically set the field if the `os.name` correspond to RHEL 6-8?

Initially, I only modified standalone.sh and domain.sh. @jamezp had mentioned that we'd need launcher updates as well. @jamezp Did I misunderstand what you meant?

**Comment by fjuma** [comment]
Just FYI, after more discussions on this approach, am going to investigate fixing the underlying issue in JBoss Modules directly instead of here so will likely close this PR.

File: launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java (hunk 1 above)
Comment (jamezp) [reply launcher/src/main/java/org/wildfly/core/launcher/AbstractCommandBuilder.java:614]: @fjuma I just meant we should probably do what we're doing in the bash script here. I don't think we need to expose it, just add it as a JAVA_OPT, e.g. `addJavaOption("-Djboss.modules.os-name=el8)`.

**Comment by jamezp** [comment]
The main question I have is does the host executing `domain.sh` *always* launch other hosts on the same OS? For example can the host that executes `domain.sh` run on RHEL 8, but has another host controller on RHEL 7 it launches?

**Comment by fjuma** [comment]
I've created https://issues.redhat.com/browse/MODULES-400 to track fixing the underlying issue directly in JBoss Modules instead so am going to close this PR.

**Pull request closed without merging**